Source/WebCore/Modules/WebGPU/GPUQueue.cpp

```diff
@@ -350,17 +350,17 @@ static void* copyToDestinationFormat(const uint8_t* rgbaBytes, GPUTextureFormat
     case GPUTextureFormat::R16uint:
     case GPUTextureFormat::R16sint: {
         uint16_t* data = (uint16_t*)malloc(sizeInBytes / 2);
-        for (size_t i = 0; i < sizeInBytes; i += 4)
-            data[i] = rgbaBytes[i];
+        for (size_t i = 0, i0 = 0; i < sizeInBytes; i += 4, ++i0)
+            data[i0] = rgbaBytes[i];
 
         sizeInBytes = sizeInBytes / 2;
         return data;
     }
 
     case GPUTextureFormat::R16float: {
         __fp16* data = (__fp16*)malloc(sizeInBytes / 2);
-        for (size_t i = 0; i < sizeInBytes; i += 4)
-            data[i] = rgbaBytes[i];
+        for (size_t i = 0, i0 = 0; i < sizeInBytes; i += 4, ++i0)
+            data[i0] = rgbaBytes[i];
```

Comment:
Should this be `data[i0] = rgbaBytes[i] / 255.f` to be consistent with the `Rgba16float` version:

```
    case GPUTextureFormat::Rgba16float: {
        __fp16* data = (__fp16*)malloc(sizeInBytes * sizeof(__fp16));
        for (size_t i = 0; i < sizeInBytes; ++i)
            data[i] = rgbaBytes[i] / 255.f;

        sizeInBytes = sizeInBytes * sizeof(__fp16);
        return data;
    }
```

Reply:
oh yes it should :), thank you for noticing